src/ansys/additive/core/parametric_study/parametric_study.py

```diff
@@ -1670,38 +1662,108 @@ def _test_single_bead_input(self, row: pd.Series) -> tuple[bool, str]:
         Returns
         -------
         tuple[bool, str]
-            bool, True if the single bead input is valid, False otherwise.
-            string, Error message if the single bead input is invalid.
+            bool, True if the input is valid, False otherwise.
+            string, Error message if the input is invalid.
         """
+
         try:
-            test_machine = AdditiveMachine(
+            allowed_types = [
+                SimulationType.SINGLE_BEAD,
+                SimulationType.POROSITY,
+                SimulationType.MICROSTRUCTURE,
+            ]
+            if row[ColumnNames.TYPE] in allowed_types:
```

Comment:
```suggestion
            if row[ColumnNames.TYPE] not in allowed_types:
                return (False, f"Invalid simulation type: {row[ColumnNames.TYPE]}.")
```
Avoid putting `if` statements within `if` statements. It increases the cyclomatic complexity, https://linearb.io/blog/cyclomatic-complexity.